docs/developer_guide/explanations/developer_conventions.rst

```diff
@@ -27,3 +27,166 @@ When updating dependencies or modifying `conda/meta.yaml`, it is important to te
+
+Though commits can be made using only a summary, it is recommended to also write a description, even if you think it is unnecessary because from a reviewers perspective, it is always better to have more information than less.
+
+The order of commits can be just as important as the commits themselves. Commits should be ordered in a logical manner and help tell the reviewer or another developer in the future the story of changes made, which may be slightly different from the order you made them in.
+
+A good reference for writing commits is the GitHub blog post `"Write Better Commits, Build Better Projects" <https://github.blog/developer-skills/github/write-better-commits-build-better-projects/>`_ which will walk you through both how to write better commits, but also how to easily re-order commits using `interactive rebase <https://docs.github.com/en/get-started/using-git/about-git-rebase>`_.
+
+Now is a good time to mention that it's important not to make commits too big. Not only are small, concise commits easier to review and understand, they also make it easier to revert if necessary.
+
+Pull Requests
+~~~~~~~~~~~~~
+
+When you have completed your work on a new feature or bug fix, We ask that you check your work thoroughly to make sure it works functionally as you expect it too and that tests pass on your machine locally. Once you are happy with your branch, you should create a pull request (PR). There are two types of PR, a draft PR and a regular PR.
+
+A draft PR can be useful if you wish to start a conversation with collaborators as soon as your work has reached a point where you would like feedback. It's important to note that once you make a draft PR, all changes pushed to the remote development branch will trigger Mantid Imagings `GitHub workflows <https://github.com/mantidproject/mantidimaging/actions>`_, which will run static analysis checks, run all unit tests, screenshot tests, and finally build the documentation. This can be useful to see if your changes have broken anything, though running them is time consuming.
+Only make a draft PR when you are as close as possible to finishing your work if you haven't already. This will save on compute time by not triggering the workflows on every push. This will also help avoid any confusion as to whether the PR is ready for review or not - though if your PR is a draft, it should be assumed that it is not yet ready for review.
+
+Once you are happy with your work and are ready for it to be reviewed, you should convert the draft PR to a regular PR (skipping the draft PR stage is also fine). Make sure to assign yourself to the PR if you haven't already under "Assignees" (not as the reviewer). Then move your issue to the "Review" column on the `project board <https://github.com/orgs/mantidproject/projects/13/views/6?>`_ your issue is associated with, which will let the Mantid Imaging team know that your PR is ready for review.
```

Comment:
Maybe change to `Once you are happy with your work, it passes tests and you are ready for it to be reviewed ...`

Reply:
I agree, this is clearer in stating what we require before a PR is ready for review.